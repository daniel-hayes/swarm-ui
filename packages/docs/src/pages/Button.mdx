import { Button } from 'swarm-components'

# Button
This is a button for swarm 2
```.jsx
<Button>Click</Button>
```
```.jsx
<Button primary>Click</Button>
```
```.jsx
<Button neutral>Click</Button>
```
```.jsx
<Button bordered>Click</Button>
```
```.jsx
<Button disabled>Click</Button>
```

<Button bordered>Click</Button>

# Inverted
<div style={{backgroundColor: 'black'}}>
    <Button inverted>Inverted</Button>
</div>
<div style={{backgroundColor: 'green'}}>
    <Button inverted>Inverted</Button>
</div>

# Button V2
<Button primary>Hello</Button>
# Button V1
<button class="button button-primary">old</button>