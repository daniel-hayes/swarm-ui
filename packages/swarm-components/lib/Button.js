function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import * as React from 'react';
import Icon from './Icon';

/**
 *
 * @param {*} props
 *
 */
export const getButtonType = props => {
  let buttonType = 'default';
  if (props.disabled) buttonType = 'disabled';
  if (props.primary) buttonType = 'primary';
  if (props.neutral) buttonType = 'neutral';
  if (props.bordered) buttonType = 'bordered';
  if (props.inverted) buttonType = 'inverted';
  return buttonType;
};
export const getSwarmSize = props => {
  let size = 'default';
  if (props.small) size = 'small';
  return size;
};
export const getIconPosition = props => {
  let position = 'left';
  if (props.right) position = 'right';
  if (!props.children) position = 'only';
  return position;
}; // TODO: find a better, more dynamic solution

const FILLS = {
  default: 'var(--color-viridian)',
  disabled: 'var(--color-gray-4)',
  primary: 'var(--color-white)',
  neutral: 'var(--color-gray-7)',
  bordered: 'var(--color-gray-7)',
  inverted: 'var(--color-white)'
};

const Button = props => {
  // destructuring to not pass invalid attributes to node
  const {
    primary,
    neutral,
    bordered,
    inverted,
    small,
    icon,
    right,
    children,
    ...rest
  } = props;

  if (props.disabled && props.onClick) {
    delete props.onClick;
  }

  const buttonType = getButtonType(props);
  return React.createElement("button", _extends({
    "data-swarm-button": buttonType,
    "data-swarm-size": getSwarmSize(props),
    "data-icon": getIconPosition(props)
  }, rest), icon ? React.createElement("span", null, right ? React.createElement(React.Fragment, null, children, React.createElement(Icon, {
    shape: icon,
    size: "xs",
    color: FILLS[buttonType]
  })) : React.createElement(React.Fragment, null, React.createElement(Icon, {
    shape: icon,
    size: "xs",
    color: FILLS[buttonType]
  }), children)) : children);
};

export default Button;
Button.__docgenInfo = {
  "description": "",
  "methods": [],
  "displayName": "Button",
  "props": {
    "bordered": {
      "required": false,
      "flowType": {
        "name": "boolean"
      },
      "description": "The bordered button"
    },
    "className": {
      "required": false,
      "flowType": {
        "name": "string"
      },
      "description": "Classes for additional styles to be applied"
    },
    "disabled": {
      "required": false,
      "flowType": {
        "name": "boolean"
      },
      "description": "Indicates whether the button is disabled or not"
    },
    "icon": {
      "required": false,
      "flowType": {
        "name": "string"
      },
      "description": "Supports icon classes found at https://meetup.github.io/swarm-icons/"
    },
    "inverted": {
      "required": false,
      "flowType": {
        "name": "boolean"
      },
      "description": "Inverted styles for dark backgrounds"
    },
    "neutral": {
      "required": false,
      "flowType": {
        "name": "boolean"
      },
      "description": "The neutral style"
    },
    "onClick": {
      "required": false,
      "flowType": {
        "name": "signature",
        "type": "function",
        "raw": "(*) => void",
        "signature": {
          "arguments": [{
            "name": "",
            "type": {
              "name": "unknown"
            }
          }],
          "return": {
            "name": "void"
          }
        }
      },
      "description": "The function invoked when interacting with Button"
    },
    "primary": {
      "required": false,
      "flowType": {
        "name": "boolean"
      },
      "description": "The primary style"
    },
    "right": {
      "required": false,
      "flowType": {
        "name": "boolean"
      },
      "description": "Aligns the icon to the right"
    },
    "small": {
      "required": false,
      "flowType": {
        "name": "boolean"
      },
      "description": "Use the small button size"
    },
    "children": {
      "required": true,
      "flowType": {
        "name": "ReactNode",
        "raw": "React.Node"
      },
      "description": ""
    }
  }
};