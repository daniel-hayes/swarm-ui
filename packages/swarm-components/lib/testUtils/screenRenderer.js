"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _hapi = _interopRequireDefault(require("hapi"));

var _path = _interopRequireDefault(require("path"));

var _inert = _interopRequireDefault(require("inert"));

var _puppeteer = _interopRequireDefault(require("puppeteer"));

var _server = _interopRequireDefault(require("react-dom/server"));

var screenRenderer =
/*#__PURE__*/
function () {
  function screenRenderer(config) {
    (0, _classCallCheck2["default"])(this, screenRenderer);
    this.config = (0, _objectSpread2["default"])({
      viewport: {
        width: 800,
        height: 600
      },
      verbose: false,
      port: 4000,
      padding: '1em'
    }, config);
    this.routeIndex = 0;
  }

  (0, _createClass2["default"])(screenRenderer, [{
    key: "log",
    value: function log() {
      if (this.config.verbose) {
        var _console;

        (_console = console).log.apply(_console, arguments);
      }
    }
  }, {
    key: "init",
    value: function () {
      var _init = (0, _asyncToGenerator2["default"])(
      /*#__PURE__*/
      _regenerator["default"].mark(function _callee() {
        var _this$config, port, staticPath;

        return _regenerator["default"].wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                _this$config = this.config, port = _this$config.port, staticPath = _this$config.staticPath;
                _context.next = 3;
                return _puppeteer["default"].launch();

              case 3:
                this.browser = _context.sent;
                this.log('Browser is running');
                this.server = _hapi["default"].server({
                  port: port
                });
                _context.next = 8;
                return this.server.register(_inert["default"]);

              case 8:
                // serving static files
                this.server.route({
                  method: 'GET',
                  path: '/static/{param*}',
                  handler: {
                    directory: {
                      path: _path["default"].join(__dirname, staticPath)
                    }
                  }
                });
                _context.next = 11;
                return this.server.start();

              case 11:
                this.log('Server running on %s', this.server.info.uri);
                return _context.abrupt("return", this);

              case 13:
              case "end":
                return _context.stop();
            }
          }
        }, _callee, this);
      }));

      function init() {
        return _init.apply(this, arguments);
      }

      return init;
    }()
  }, {
    key: "createRoute",
    value: function createRoute(slug, element) {
      var _this = this;

      var links = this.config.stylesheets.map(function (stylesheet) {
        return "<link\n\t\t\t\t\t\trel=\"stylesheet\"\n\t\t\t\t\t\ttype=\"text/css\"\n\t\t\t\t\t\thref=\"/static/".concat(stylesheet, "\" />");
      }).join('\n');
      return {
        method: 'GET',
        path: "/".concat(slug),
        handler: function handler() {
          return "<html>\n\t\t\t\t\t\t\t\t<head>\n\t\t\t\t\t\t\t\t\t".concat(links, "\n\t\t\t\t\t\t\t\t</head>\n\t\t\t\t\t\t\t\t<body style=\"padding: ").concat(_this.config.padding, "\">\n\t\t\t\t\t\t\t\t").concat(_server["default"].renderToStaticMarkup(element), "\n\t\t\t\t\t\t\t\t</body>\n\t\t\t\t\t\t\t</html>");
        }
      };
    }
  }, {
    key: "stop",
    value: function () {
      var _stop = (0, _asyncToGenerator2["default"])(
      /*#__PURE__*/
      _regenerator["default"].mark(function _callee2() {
        return _regenerator["default"].wrap(function _callee2$(_context2) {
          while (1) {
            switch (_context2.prev = _context2.next) {
              case 0:
                _context2.next = 2;
                return this.browser.close();

              case 2:
                return _context2.abrupt("return", this.server.stop());

              case 3:
              case "end":
                return _context2.stop();
            }
          }
        }, _callee2, this);
      }));

      function stop() {
        return _stop.apply(this, arguments);
      }

      return stop;
    }()
  }, {
    key: "screenshot",
    value: function () {
      var _screenshot = (0, _asyncToGenerator2["default"])(
      /*#__PURE__*/
      _regenerator["default"].mark(function _callee3(element, screenshotConfig) {
        var page, slug, testUrl;
        return _regenerator["default"].wrap(function _callee3$(_context3) {
          while (1) {
            switch (_context3.prev = _context3.next) {
              case 0:
                _context3.next = 2;
                return this.browser.newPage();

              case 2:
                page = _context3.sent;
                page.setViewport(screenshotConfig && screenshotConfig.viewport || this.config.viewport);
                slug = "route-".concat(this.routeIndex++);
                this.server.route(this.createRoute(slug, element));
                testUrl = "".concat(this.server.info.uri, "/").concat(slug);
                this.log("Testing: ".concat(testUrl));
                _context3.next = 10;
                return page["goto"](testUrl);

              case 10:
                return _context3.abrupt("return", page.screenshot());

              case 11:
              case "end":
                return _context3.stop();
            }
          }
        }, _callee3, this);
      }));

      function screenshot(_x, _x2) {
        return _screenshot.apply(this, arguments);
      }

      return screenshot;
    }()
  }]);
  return screenRenderer;
}();

var _default =
/*#__PURE__*/
function () {
  var _ref = (0, _asyncToGenerator2["default"])(
  /*#__PURE__*/
  _regenerator["default"].mark(function _callee4(config) {
    return _regenerator["default"].wrap(function _callee4$(_context4) {
      while (1) {
        switch (_context4.prev = _context4.next) {
          case 0:
            return _context4.abrupt("return", new screenRenderer(config).init());

          case 1:
          case "end":
            return _context4.stop();
        }
      }
    }, _callee4);
  }));

  return function (_x3) {
    return _ref.apply(this, arguments);
  };
}();

exports["default"] = _default;