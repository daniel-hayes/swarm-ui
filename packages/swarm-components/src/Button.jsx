// @flow
import * as React from 'react';
import Icon from './Icon';

type Props = {
	/**
	* The bordered button
	*/
	bordered?: boolean,
	/**
	* Classes for additional styles to be applied
	*/
	className?: string,
	/**
	* Indicates whether the button is disabled or not
	*/
	disabled?: boolean,
	/**
	* Supports icon classes found at https://meetup.github.io/swarm-icons/
	*/
	icon?: string,
	/**
	* Inverted styles for dark backgrounds
	*/
	inverted?: boolean,
	/**
	* The neutral style
	*/
	neutral?: boolean,
	/**
	* The function invoked when interacting with Button
	*/
	onClick?: (*) => void,
	/**
	* The primary style
	*/
	primary?: boolean,
	/**
	* Aligns the icon to the right
	*/
	right?: boolean,
	/**
	* Use the small button size
	*/
	small?: boolean,
	children: React.Node,
};

/**
 *
 * @param {*} props
 *
 */
export const getButtonType = (props: Props): string => {
	let buttonType = 'default';
	if (props.disabled) buttonType = 'disabled';
	if (props.primary) buttonType = 'primary';
	if (props.neutral) buttonType = 'neutral';
	if (props.bordered) buttonType = 'bordered';
	if (props.inverted) buttonType = 'inverted';
	return buttonType;
};

export const getSwarmSize = (props: Props): string => {
	let size = 'default';
	if (props.small) size = 'small';
	return size;
};

export const getIconPosition = (props: Props): string => {
	let position = 'left';
	if (props.right) position = 'right';
	if (!props.children) position = 'only';
	return position;
};

// TODO: find a better, more dynamic solution
const FILLS = {
	default: 'var(--color-viridian)',
	disabled: 'var(--color-gray-4)',
	primary: 'var(--color-white)',
	neutral: 'var(--color-gray-7)',
	bordered: 'var(--color-gray-7)',
	inverted: 'var(--color-white)',
};

const Button = (props: Props): React.Element<'button'> => {
	// destructuring to not pass invalid attributes to node
	const {
		primary,
		neutral,
		bordered,
		inverted,
		small,
		icon,
		right,
		children,
		...rest
	} = props;

	if (props.disabled) {
		delete props.onClick;
	}

	const buttonType = getButtonType(props);

	return (
		<button
			data-swarm-button={buttonType}
			data-swarm-size={getSwarmSize(props)}
			data-icon={getIconPosition(props)}
			{...rest}
		>
			{icon ? (
				<span>
					{right ? (
						<>
							{children}
							<Icon shape={icon} size="xs" color={FILLS[buttonType]} />
						</>
					) : (
						<>
							<Icon shape={icon} size="xs" color={FILLS[buttonType]} />
							{children}
						</>
					)}
				</span>
			) : (
				children
			)}
		</button>
	);
};

export default Button;
