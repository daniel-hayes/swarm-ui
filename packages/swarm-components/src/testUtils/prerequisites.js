import fs from 'fs';
import Path from 'path';
import { Exception } from 'handlebars';

// setup.js
module.exports = async () => {
	if (
		!fs.existsSync(
			Path.join(__dirname, '../../../swarm-styles/dist/global.css')
		)
	) {
		throw new Exception(
			"Can't find packages/swarm-styles/dist/global.css! Did you run `yarn build`?"
		);
	}

	if (
		!fs.existsSync(Path.join(__dirname, '../../../swarm-styles/dist/main.css'))
	) {
		throw new Exception(
			"Can't find packages/swarm-styles/dist/main.css! Did you run `yarn build`?"
		);
	}

	if (
		!fs.existsSync(
			Path.join(__dirname, '../../../swarm-docs/src/assets/graphik.css')
		)
	) {
		throw new Exception(
			"Can't find packages/swarm-docs/src/assets/graphik.css"
		);
	}
};
