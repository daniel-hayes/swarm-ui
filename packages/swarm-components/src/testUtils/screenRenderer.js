import Hapi from 'hapi';
import Path from 'path';
import inert from 'inert';

import puppeteer from 'puppeteer';

import ReactDOMServer from 'react-dom/server';

class screenRenderer {
	constructor(config) {
		this.config = {
			viewport: { width: 800, height: 600 },
			verbose: false,
			port: 4000,
			host: 'localhost',
			padding: '1em',
			...config,
		};

		this.routeIndex = 0;
	}

	log(...params) {
		if (this.config.verbose) {
			console.log(...params);
		}
	}

	async init() {
		const { port, host, staticPath } = this.config;

		this.browser = await puppeteer.launch();
		this.log('Browser is running');

		this.server = Hapi.server({
			port,
			host,
		});

		await this.server.register(inert);

		// serving static files
		this.server.route({
			method: 'GET',
			path: '/static/{param*}',
			handler: {
				directory: {
					path: Path.join(__dirname, staticPath),
				},
			},
		});

		await this.server.start();
		this.log('Server running on %s', this.server.info.uri);

		return this;
	}

	createRoute(slug, element, bodyStyle='') {
		const links = this.config.stylesheets
			.map(
				stylesheet =>
					`<link
						rel="stylesheet"
						type="text/css"
						href="/static/${stylesheet}" />`
			)
			.join('\n');

		return {
			method: 'GET',
			path: `/${slug}`,
			handler: () => `<html>
								<head>
									${links}
								</head>
								<body style="padding: ${this.config.padding}; ${bodyStyle}">
								${ReactDOMServer.renderToStaticMarkup(element)}
								</body>
							</html>`,
		};
	}

	async stop() {
		await this.browser.close();
		return this.server.stop();
	}

	async screenshot(element, screenshotConfig) {
		const page = await this.browser.newPage();
		page.setViewport(
			(screenshotConfig && screenshotConfig.viewport) || this.config.viewport
		);

		const slug = `route-${this.routeIndex++}`;

		this.server.route(this.createRoute(slug, element, screenshotConfig.bodyStyle));

		const testUrl = `${this.server.info.uri}/${slug}`;

		this.log(`Testing: ${testUrl}`);
		await page.goto(testUrl);

		return page.screenshot();
	}
}

export default async config => new screenRenderer(config).init();
