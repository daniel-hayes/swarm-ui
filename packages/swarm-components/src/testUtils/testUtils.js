import React from 'react';

export const getTestMarkup = (testCases) => (
    <div>
        {testCases.map((test, index) => {
            const label = test[0];
            const component = test[1];
            const wrapperStyle = test[2];

            return (
                <div key={index}>
                    <p>{label}</p>
                    <div style={wrapperStyle ? wrapperStyle : {}}>
                        {component}
                    </div>
                    <br/>
                </div>
            );
        })}
    </div>
);
