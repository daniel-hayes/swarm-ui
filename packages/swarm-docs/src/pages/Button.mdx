import { Button } from '@meetup/swarm-components'
import { Icon } from '@meetup/swarm-components'

# Button
This is a button for swarm 2
```.jsx
<Button>Click</Button>
```
```.jsx
<Button primary>Click</Button>
```
```.jsx
<Button neutral>Click</Button>
```
```.jsx
<Button bordered>Click</Button>
```
```.jsx
<Button disabled>Click</Button>
```

<Button bordered>Click</Button>

# Inverted
<div style={{backgroundColor: 'black'}}>
    <Button inverted>Inverted</Button>
</div>
<div style={{backgroundColor: 'green'}}>
    <Button inverted>Inverted</Button>
</div>

# Button V2
<Button primary icon='arrow-right' right>Hello</Button>
<Button primary icon='arrow-right'>Hello</Button>
<Button primary icon='arrow-right' />

# Icon
<Icon shape="arrow-right" />

# Button V1
<button class="button button-primary">old</button>