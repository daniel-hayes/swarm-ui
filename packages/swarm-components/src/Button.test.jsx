import React from 'react';

import { toMatchImageSnapshot } from 'jest-image-snapshot';
expect.extend({ toMatchImageSnapshot });

import getScreenRenderer from './testUtils/screenRenderer';
import Button from './Button';
import { getTestMarkup } from './testUtils/testUtils.js';

describe('Button', () => {
	let renderer;

	// This is ran when the suite starts up.
	beforeAll(async () => {
		renderer = await getScreenRenderer({
			viewport: { width: 300, height: 2000 },
			// verbose: true,
		});
	});

	// This is ran when the suite is done - stop the renderer.
	afterAll(() => {
		// comment next line out if you want to open it in your browser for debugging
		return renderer.stop();
	});

	const testCases = [
		['Default', <Button key="1" >Press me</Button>],
		['Disabled', <Button disabled key="2">Can&apos;t press me</Button>],
		['Primary', <Button primary key="3">Must press me</Button>],
		['Bordered', <Button bordered key="4">Press me</Button>],
		['Icon with text', <Button icon="bolt" key="5">Press me</Button>],
		['Icon (right) with text', <Button right icon="chevron-right" key="6">Press me</Button>],
		['Icon without text', <Button icon="bolt" key="7" />],
		['Reset', <Button reset key="8"> Press me</Button>],
		['Inverted', <Button inverted key="9">Press me</Button>, {backgroundColor: 'black', padding: '1em'}],
		['Neutral', <Button neutral key="10">Press me</Button>],
		['Small', <Button small key="11">Press me</Button>],
		['Grow', <Button grow key="12">Full Width Button</Button>],
	];

	it('Default', async () => {
		expect(
			await renderer.screenshot(getTestMarkup(testCases))
		).toMatchImageSnapshot();
	});
});
