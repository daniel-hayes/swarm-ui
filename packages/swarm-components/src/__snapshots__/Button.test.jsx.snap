// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot: Bordered 1`] = `
<button
  data-icon="left"
  data-swarm-button="bordered"
  data-swarm-size="default"
  data-swarm-width="default"
>
  Press me
</button>
`;

exports[`Button Snapshot: Default 1`] = `
<button
  data-icon="left"
  data-swarm-button="default"
  data-swarm-size="default"
  data-swarm-width="default"
>
  Press me
</button>
`;

exports[`Button Snapshot: Disabled 1`] = `
<button
  data-icon="left"
  data-swarm-button="disabled"
  data-swarm-size="default"
  data-swarm-width="default"
  disabled={true}
>
  Can't press me
</button>
`;

exports[`Button Snapshot: Grow 1`] = `
<button
  data-icon="left"
  data-swarm-button="default"
  data-swarm-size="default"
  data-swarm-width="grow"
>
  Full Width Button
</button>
`;

exports[`Button Snapshot: Icon (right) with text 1`] = `
<button
  data-icon="right"
  data-swarm-button="default"
  data-swarm-size="default"
  data-swarm-width="default"
>
  <span>
    Press me
    <Icon
      color="var(--color-viridian)"
      shape="chevron-right"
      size="xs"
    />
  </span>
</button>
`;

exports[`Button Snapshot: Icon with text 1`] = `
<button
  data-icon="left"
  data-swarm-button="default"
  data-swarm-size="default"
  data-swarm-width="default"
>
  <span>
    <Icon
      color="var(--color-viridian)"
      shape="bolt"
      size="xs"
    />
    Press me
  </span>
</button>
`;

exports[`Button Snapshot: Icon without text 1`] = `
<button
  data-icon="only"
  data-swarm-button="default"
  data-swarm-size="default"
  data-swarm-width="default"
>
  <span>
    <Icon
      color="var(--color-viridian)"
      shape="bolt"
      size="xs"
    />
  </span>
</button>
`;

exports[`Button Snapshot: Inverted 1`] = `
<button
  data-icon="left"
  data-swarm-button="inverted"
  data-swarm-size="default"
  data-swarm-width="default"
>
  Press me
</button>
`;

exports[`Button Snapshot: Neutral 1`] = `
<button
  data-icon="left"
  data-swarm-button="neutral"
  data-swarm-size="default"
  data-swarm-width="default"
>
  Press me
</button>
`;

exports[`Button Snapshot: Primary 1`] = `
<button
  data-icon="left"
  data-swarm-button="primary"
  data-swarm-size="default"
  data-swarm-width="default"
>
  Must press me
</button>
`;

exports[`Button Snapshot: Reset 1`] = `
<button
  data-icon="left"
  data-swarm-button="reset"
  data-swarm-size="default"
  data-swarm-width="default"
>
   Press me
</button>
`;

exports[`Button Snapshot: Small 1`] = `
<button
  data-icon="left"
  data-swarm-button="default"
  data-swarm-size="small"
  data-swarm-width="default"
>
  Press me
</button>
`;
